### ISSUE TITLE
[$1000] Web - Split bill icons are aligned to the left

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#22993]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Split bill with 8 users
2. See the icons on the split bill report.

## Expected Result:
 Split bill icons are aligned to the right

## Actual Result:
Split bill icons are aligned to the left

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.40-5
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

![image (31)](https://github.com/Expensify/App/assets/93399543/a0dc4463-e0bb-4a4b-a3ef-a60881cd0fa2)
![image (30)](https://github.com/Expensify/App/assets/93399543/536fd5a2-2415-4c55-9ce6-fe1c9511d502)
![image (29)](https://github.com/Expensify/App/assets/93399543/9c9a7703-da9e-42f2-b59a-402c0a693425)

**Expensify/Expensify Issue URL:**
**Issue reported by:** @daveSeife
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1689299771028809

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~016faf5a12cb59c166</li>
        <li>Upwork Job ID: 1681073914064621568</li>
        <li>Last Price Increase: 2023-07-17</li>
    </ul>
</details>

**Comment by Jahanzaib00** [comment]
Hello, from last 2 days I've been working on some of the issues but no matter how quickly I resolve them or send the proposal. Somehow, I've never assigned the issue formally. 
Can some guide, what's the issue?

**Comment by jayeshmangwani** [comment]
The same root cause as this https://github.com/Expensify/App/issues/22632 issue

cc: @jliexpensify

**Comment by aditya-ragi** [comment]
@jayeshmangwani does this mean the issue is closed and don't need any work?

**Comment by DrLoopFall** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Split bill icons are not aligned properly

### What is the root cause of that problem?
We are using `left` for positioning/shifting avatars, which causes it to use more space than required.
https://github.com/Expensify/App/blob/702d8e254760b3ac3d90586d80528acebbb0b529/src/styles/StyleUtils.js#L906
https://github.com/Expensify/App/blob/702d8e254760b3ac3d90586d80528acebbb0b529/src/styles/StyleUtils.js#L923
Also, currently, we are calculating the width, as a [workaround](https://github.com/Expensify/App/issues/22445#issuecomment-1629454284), which is also a part of the problem in this issue.
> @stitesExpensify I think the `width` was set as a workaround for the issue described in my [proposal](https://github.com/Expensify/App/issues/22445#issuecomment-1627413019) (the 2nd issue)
> 
> > We are using `left` for positioning/shifting avatars but the `PressableWithoutFeedback` itself is not shifted, which causes it to be clickable in both the original and shifted positions.
> 
> If we use `margin-left` instead, we can remove the `width`, which may prevent any future issues.

https://github.com/Expensify/App/blob/702d8e254760b3ac3d90586d80528acebbb0b529/src/components/MultipleAvatars.js#L138-L144

### What changes do you think we should make in order to solve the problem?
Use `margin-left` instead of `left` for positioning the avatar.
```js
// in getHorizontalStackedAvatarStyle
- left: -(overlapSize * index)
+ marginLeft: index > 0 ? -overlapSize : 0

// in getHorizontalStackedOverlayAvatarStyle
- left: -(oneAvatarSize.width * 2 + oneAvatarBorderWidth * 2)
+ marginLeft: -(oneAvatarSize.width + oneAvatarBorderWidth * 2)
```
We should also need to remove the width, as we don't need it anymore, because this proposal also removes the need for the [workaround](https://github.com/Expensify/App/issues/22445#issuecomment-1629454284) as it fixes the issue in the root, and removing it may prevent future issues like this.
https://github.com/Expensify/App/blob/702d8e254760b3ac3d90586d80528acebbb0b529/src/components/MultipleAvatars.js#L138-L144

**Comment by DrLoopFall** [comment]
I've already made proposals in issue #22632 and #22685 which also has the same exact root cause as this one, implementing any one of the proposals would solve all these 3 issues.

Also, I've made a proposal above just in case.

@puneetlath please have a look here also.

**Comment by jliexpensify** [comment]
Hi @Santhosh-Sellavel - we've had a few comments stating that this issue has the [same root cause](https://github.com/Expensify/App/issues/22993#issuecomment-1637978597) as a [few others](https://github.com/Expensify/App/issues/22993#issuecomment-1638179117). Could you check this out, and let us know if this is the case? If so, should we wait on a PR from another issue to resolve this one?

**Comment by jliexpensify** [comment]
Bump @Santhosh-Sellavel !

**Comment by Santhosh-Sellavel** [comment]
Missed this one somehow!

**Comment by Santhosh-Sellavel** [comment]
@jliexpensify This seems like a dupe of #22632 & the same RC.

We can close this one straightaway, and nothing is due here!
